src/handlers/miner_hbbft_handler.erl

```diff
@@ -240,9 +250,43 @@ handle_message(BinMsg, Index, State) when is_binary(BinMsg) ->
             lager:warning("got truncated message: ~p:", [BinMsg]),
             ignore
     end;
-handle_message(Msg, Index, State=#state{hbbft = HBBFT}) ->
+handle_message(Msg, Index, State=#state{hbbft = HBBFT, skip_votes = Skips}) ->
     Round = hbbft:round(HBBFT),
     case Msg of
+        {proposed_skip, ProposedRound, IndexRound} when ProposedRound =/= Round ->
+            lager:info("proposed skip to round ~p", [ProposedRound]),
+            case process_skips(ProposedRound, IndexRound, State#state.f, Index, Skips) of
+                {skip, Skips1} ->
+                    lager:info("skipping"),
+                    %% ask for some time
+                    miner:reset_late_block_timer(),
+                    SkipGossip = {multicast, t2b({proposed_skip, ProposedRound, Round})},
+                    %% skip but don't discard rounds until we get a clean round
+                    {HBBFT1, ToSend1} =
+                        case hbbft:next_round(HBBFT, ProposedRound, []) of
+                            {H1, ok} ->
+                                {H1, []};
+                            {H1, {send, NextMsgs}} ->
+                                {H1, NextMsgs}
+                        end,
+                    {HBBFT2, ToSend2} =
+                        case hbbft:start_on_demand(HBBFT1) of
+                            {H2, already_started} ->
+                                {H2, []};
+                            {H2, {send, Msgs}} ->
+                                {H2, Msgs}
+                        end,
+                    ToSend = fixup_msgs(ToSend1 ++ ToSend2),
+                    {State#state{hbbft=HBBFT2, signatures=[], artifact=undefined, sig_phase=unsent,
+                                 bba = <<>>, seen = #{}, skip_votes = Skips1},
+                     [ new_epoch , SkipGossip ] ++ ToSend};
```

Comment:
Is this not supposed to be tagged multicast here for `{new_epoch, SkipGossip}`?

Reply:
no, new epoch is an command to relcast, not a message